test/functional/p2p_filter.py

```diff
@@ -135,14 +136,38 @@ def test_msg_mempool(self):
         self.log.info("Check that a node with bloom filters enabled services p2p mempool messages")
         filter_peer = P2PBloomFilter()
 
-        self.log.debug("Create a tx relevant to the peer before connecting")
-        txid, _ = self.wallet.send_to(from_node=self.nodes[0], scriptPubKey=filter_peer.watch_script_pubkey, amount=9 * COIN)
+        self.log.info("Create two tx before connecting, one relevant to the node another that is not")
+        rel_txid, _ = self.wallet.send_to(from_node=self.nodes[0], scriptPubKey=filter_peer.watch_script_pubkey, amount=1 * COIN)
+        rel_wtxid = self.nodes[0].getmempoolentry(rel_txid)["wtxid"]
+        irr_txid, _ = self.wallet.send_to(from_node=self.nodes[0], scriptPubKey=getnewdestination()[1], amount=2 * COIN)
+        irr_wtxid = self.nodes[0].getmempoolentry(irr_txid)["wtxid"]
 
-        self.log.debug("Send a mempool msg after connecting and check that the tx is received")
+        self.log.info("Send a mempool msg after connecting and check that the relevant tx is received")
         self.nodes[0].add_p2p_connection(filter_peer)
         filter_peer.send_and_ping(filter_peer.watch_filter_init)
-        filter_peer.send_message(msg_mempool())
-        filter_peer.wait_for_tx(txid)
+        filter_peer.send_and_ping(msg_mempool())
+        filter_peer.wait_for_tx(rel_txid)
+
+        self.log.info("Request the irrelevant transaction even though it has not being offered to us")
+        filter_peer.tx_received = False
+        filter_peer.send_and_ping(msg_getdata([CInv(t=MSG_WTX, h=int(irr_wtxid, 16))]))
+        assert not filter_peer.tx_received
+
+        self.log.info("Create a third transaction (that was therefore not part of the mempool message) and try to retrieve it")
+        add_txid, _ = self.wallet.send_to(from_node=self.nodes[0], scriptPubKey=getnewdestination()[1], amount=3 * COIN)
+        filter_peer.send_and_ping(msg_getdata([CInv(t=MSG_WTX, h=int(self.nodes[0].getmempoolentry(add_txid)["wtxid"], 16))]))
+        assert not filter_peer.tx_received
+
+        self.log.info("Check this is also sound for a peer that has a match-all filter set")
+        unfilter_peer = P2PBloomFilter()
+        self.nodes[0].add_p2p_connection(unfilter_peer)
+        unfilter_peer.send_and_ping(msg_mempool())
+
+        self.log.info("Here the all transactions are relevant")
```

Comment:
I think this log line could be removed as this seems part of the previous check?